server/src/test/java/org/elasticsearch/cluster/NoMasterNodeIT.java

```diff
@@ -53,104 +58,113 @@
 public class NoMasterNodeIT extends ESIntegTestCase {
 
     @Override
-    protected Settings nodeSettings(int nodeOrdinal) {
-        return Settings.builder().put(super.nodeSettings(nodeOrdinal))
-            .put(TestZenDiscovery.USE_ZEN2.getKey(), false) // tests here need adaption for Zen2
-            .build();
+    protected int numberOfReplicas() {
+        return 2;
     }
 
     public void testNoMasterActions() throws Exception {
         Settings settings = Settings.builder()
-                .put("action.auto_create_index", true)
-                .put("discovery.zen.minimum_master_nodes", 2)
-                .put(ZenDiscovery.PING_TIMEOUT_SETTING.getKey(), "200ms")
-                .put("discovery.initial_state_timeout", "500ms")
-                .put(DiscoverySettings.NO_MASTER_BLOCK_SETTING.getKey(), "all")
-                .build();
+            .put(AutoCreateIndex.AUTO_CREATE_INDEX_SETTING.getKey(), true)
+            .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), Integer.MAX_VALUE)
+            .put(GatewayService.RECOVER_AFTER_MASTER_NODES_SETTING.getKey(), 3)
```

Comment:
why add this setting? This should not be needed?

Reply:
It defaults to whatever `minimum_master_nodes` is. I could just set that to 3.

Reply:
ah that's annoying, and does not make any sense in Zen2. Can you perhaps change GatewayService to not do this if `discovery instanceof Coordinator`?

Reply:
Ok, I pushed b0e9643e686b262f65697bbd2d8f111cfc6e18e9.